scripts/select-java

```diff
@@ -1,43 +1,55 @@
 #!/bin/bash
 
-has_java=false
+function select_java_debian() {
+    for version in "11" "8"; do
+        pkg_name=openjdk-${version}-jre-headless
+        if ! dpkg-query --status ${pkg_name}; then
+            continue
+        fi
+        java=$(dpkg -L ${pkg_name} | grep '/java$')
+        if [ -n "$java" ]; then
+            echo $java
+            return
+        fi
+    done
+}
+
+function select_java_fedora() {
+    for version in "11" "1.8.0"; do
+        pkg_name=java-${version}-openjdk-headless
+        if ! rpm --quiet -q ${pkg_name}; then
+            continue
+        fi
+        java=$(rpm -ql ${pkg_name} | grep '/java$')
+        if [ -n "$java" ]; then
+            echo $java
+            return
+        fi
+    done
+}
+
+function select_java_others() {
+    javaver=$(/usr/bin/java -version 2>&1|head -n1|cut -f 3 -d " ")
+    if [[ "$javaver" =~ ^\"1.8.0 || "$javaver" =~ ^\"11.0. ]]; then
+        echo /usr/bin/java
```

Comment:
By the way, instead of /usr/bin/java, you could have just used "java" (without a path), and assume that the user has it somewhere in the path. If you want the full path, you can use `which java`. But in any case, I agree /usr/bin/java will be the common case.

Reply:
Ahh, I didn’t intend to change this part. But yes, `java` would be better. We can fix it in a follow-up change.